src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/IndirectionNode.cs

```diff
@@ -26,7 +26,7 @@ public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         public int Offset => 0;
 
         public override ObjectNodeSection Section => ObjectNodeSection.DataSection;
-        public override bool IsShareable => false;
+        public override bool IsShareable => true;
```

Comment:
Having general-purpose, shared, writable (in .data section instead of .rdata) indirection nodes seems like a recipe for future problems. Does `_indirectedNode` replaced at runtime?  It might be a good idea to make it read-only if not.